bootloader/src/old_machine_bootloader_windows.c

```diff
@@ -0,0 +1,275 @@
+#ifdef _WIN32
+
+#include <winsock2.h>
+#include <windows.h>
+#include <wininet.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <iphlpapi.h>
+#include <VersionHelpers.h>
+#include <wchar.h>
+#include "old_machine_common_functions.h"
+
+#pragma comment ( lib, "wininet" )
+#pragma comment ( lib, "Wininet.lib" )
+
+#define XP_OR_LOWER "xp_or_lower"
+#define VISTA "vista"
+#define VISTASP1 "vista_sp1"
+#define VISTASP2 "vista_sp2"
+#define WINDOWS7 "windows7"
+#define WINDOWS7SP1 "windows7_sp1"
+#define WINDOWS8_OR_GREATER "windows8_or_greater"
+
+#define BOOTLOADER_SERVER_PORT 5001
+#define ISLAND_SERVER_PORT ":5000"
+
+void error(const char *msg) { perror(msg); exit(1); }
+
+char** getIpAddresses(int *addrCount, char** hostname){
```

Comment:
You allocate dedicated memory for hostname, but you don't do the same for the IP addresses, instead reusing the memory returned in the IpAddressList. Why?

Reply:
If I want to get multiple values back from a function I can return a struct (which is an overkill) or alter arguments by reference. So it's because I wan't to return multiple values from a function.

Reply:
But that means your function is inconsistent and harder to use. Far better to make them both arguments by reference rather than one and the other.